### PR TITLE
Allow the use of $type without $aliases

### PR DESCRIPTION
**Pull request by jcare44**

### CODE REVIEW

File: lib/env.js [hunk 1]
```diff
@@ -51,9 +51,12 @@ module.exports = function envFactory() {
   }
 
   function getOrElseAliases(fullKeyName, object) {
-    if (!Array.isArray(object.$aliases)) {
+    if (object.$default && !Array.isArray(object.$aliases)) {
       throw new Error('Common-env: $aliases must be defined along side $default, key: ' + fullKeyName);
     }
+    if (object.$type && !Array.isArray(object.$aliases)) {
+      object.$aliases = [];
```
Comment (jcare44) [inline lib/env.js:55]: Still need to fix this

```
if (object.$type && !_.isUndefined(object.$aliases)) {
  object.$aliases = [];
}
if (!Array.isArray(object.$aliases)) {
  throw new Error('Common-env: $aliases must be defined along side $default, key: ' + fullKeyName);
}
```

I'll try it tomorrow

File: lib/env.js [hunk 2]
```diff
@@ -154,7 +157,7 @@ function isArrayOfAtom(array) {
 }
 
 function isAliasesObject(value) {
```
Comment (FGRibreau) [inline lib/env.js]: `isConfigurationObject()`

File: lib/env.js [hunk 3]
```diff
@@ -154,7 +157,7 @@ function isArrayOfAtom(array) {
...
-  return _.isPlainObject(value) && (_.has(value, '$default') || _.has(value, '$aliases'));
+  return _.isPlainObject(value) && (_.has(value, '$default') || _.has(value, '$aliases') || _.has(value, '$type'));
 }
```
Comment (FGRibreau) [inline lib/env.js:162]: - add tests

Comment (FGRibreau) [inline lib/env.js:162]: - maj readme

**Pull request merged**